Выполненный проект 6 спринта

Yuuki-that-one opened the pull request.

Yuuki-that-one committed:
Выполненный проект 6 спринта

pomegranate282922 commented on src/main/java/com/example/Lion.java:
```diff
@@ -4,29 +4,30 @@
 
 public class Lion {
 
-    boolean hasMane;
```
Ок, изоляция выполнена верно.

pomegranate282922 commented on src/test/java/TestCat.java:
```diff
@@ -0,0 +1,38 @@
+public class TestCat {
+
+    @Mock
+    private Feline feline;
```
Отлично, уместно применен мок.

pomegranate282922 commented on src/test/java/TestLionSex.java:
```diff
@@ -0,0 +1,49 @@
+            assertEquals(expected, actual);
+        } catch (Exception e) {
+            Assert.assertEquals(e.getMessage(), "Используйте допустимые значения пола животного - самец или самка");
+        }
```
Нет, так не получится. Если пробел вернёт false, тест будет зелёным, а это баг.аналогично если самец или самка выкинут исключение, тест тоже будет зелёным, а это баг.
Нужно тестирование исключения вынести из этого класса и сделать через метод assertThrows или параметр к аннотации Test.
Нужно исправить

pomegranate282922 reviewed (commented):
Отличная работа! Моки, стабы, параметризация - все по делу.
Есть важный момент в тесте на lion - нужно его поправить.

Yuuki-that-one committed:
Правки к выполненному проекту 6 спринта
Тестирование исключения lion.doesHaveMane() вынесено в класс TestLion, в  отдельный непараметризованный тест.

pomegranate282922 reviewed (approved):
Зачёт!
